### PR TITLE
EVG-7171 Slip distros with very few hosts past the throttle

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: units/crons.go [hunk 1]
```diff
@@ -808,21 +809,25 @@ func PopulateHostCreationJobs(env evergreen.Environment, part int) amboy.QueueOp
...
 		}
-		grip.Error(message.WrapError(err, message.Fields{
-			"operation": "background task creation",
-			"cron":      createHostJobName,
-			"impact":    "hosts cannot start",
-		}))
-		if err != nil {
-			return errors.WithStack(err)
+		distroCount := map[string]int{}
+		for _, h := range runningHosts {
```
Comment (ybrill) [inline units/crons.go]: It worked for me without initialization:
https://play.golang.org/p/JJNF_1J0Hoq

File: globals.go [hunk 2]
```diff
@@ -415,6 +415,7 @@ var (
 	// they're just intents, so this list omits that value.
 	ActiveStatus = []string{
 		HostRunning,
+		HostBuilding,
 		HostStarting,
 		HostProvisioning,
 		HostProvisionFailed,
```
Comment (ybrill) [inline globals.go]: Do we want hosts with `HostProvisionFailed` status included?

File: units/crons.go [hunk 3]
```diff
@@ -846,19 +851,25 @@ func PopulateHostCreationJobs(env evergreen.Environment, part int) amboy.QueueOp
...
 				//    always start spawn hosts asap
+
 			} else {
-				if submitted > env.Settings().HostInit.HostThrottle {
-					// throttle hosts, so that we're starting very
-					// few hosts on every pass. Hostinit runs very
-					// frequently, lets not start too many all at
-					// once.
-					continue
+				num, _ := distroCount[h.Distro.Id]
```
Comment (ybrill) [inline units/crons.go]: I think the _ is extra if we aren't checking whether the key is present.
https://play.golang.org/p/wlKk8YLmzCo

File: units/crons.go [hunk 4]
```diff
@@ -846,19 +851,25 @@ func PopulateHostCreationJobs(env evergreen.Environment, part int) amboy.QueueOp
...
+				if num == 0 || num < len(runningHosts)/100 {
```
Comment (ybrill) [inline units/crons.go]: are we okay with letting _all_ the requested hosts for this distro through?

Comment (ybrill) [inline units/crons.go]: but maybe this will give priority to the distros for which there's the least demand?

File: globals.go (hunk 2 above)
Comment (bsamek) [reply globals.go]: I think so. `HostProvisionFailed` can indicate a pretty wide range of things, but what it guarantees is that Evergreen has tried and failed to provision the host. Hosts in this state will eventually be terminated, so there should only ever be a small number of them.

File: units/crons.go (hunk 4 above)
Comment (bsamek) [reply units/crons.go]: I think that's inevitable. My hope is that this is conservative enough to have a reasonably small impact on high-demand distros.

**Pull request merged**